proposals/ytt/004-schema-validation/README.md

```diff
@@ -0,0 +1,1523 @@
+---
+title: "Validations"
+authors: [ "John S. Ryan <ryanjo@vmware.com>" ]
+status: "draft"
+approvers: []
+---
+
+
+# Validations
+
+## Problem Statement
+
+To date, ytt Library / Carvel Package **authors have had to "roll their own" logic to check that customer's inputs are valid.** That is: that data values are a) present/non-empty; and b) within expected range(s).
+
+Given the expressiveness of Starlark, producing the actual logic, itself, is not too onerous.
+
+However, Authors have had to...
+- write the "plumbing" code around validations (or offer their consumers a degraded UX):
+    - collecting all violations (instead of failing on first violation)
+    - ordering validation evaluation — leaves to root
+    - handling common scenarios (empty values, enums, unions, conditional)
+- hand-document validations by adding comments to their Data Values (Schema) describing those constraints.
+    - it's double the work
+    - there are now two places that need to be updated whenever a constraint changes (see [Shalloway's Law](https://netobjectivesthoughts.com/shalloways-law-and-shalloways-principle/#:~:text=%E2%80%9CWhen%20N%20things%20need%20to,%2D1%20of%20these%20things.%E2%80%9D&text=It's%20a%20law!)).
+
+**The time our users spend in developing and maintaining this code, is time lost to making progress in their first-order work.**
```

Comment:
I feel like this is the key to the whole proposal. I would make it part of main problem statement sentence

Reply:
Totally agree. Remembering: putting this sentence here and bolding it was my attempt at doing just that.  But I'm getting from you that I might have buried the lede... yeah?